crates/sncast/src/starknet_commands/script.rs

```diff
@@ -375,49 +256,53 @@ impl CairoHintProcessor<'_> {
                     },
                 ))?;
 
-                buffer
-                    .write(Felt252::from_(invoke_response.transaction_hash.0))
-                    .expect("Failed to insert transaction hash");
-
-                Ok(())
+                let res: Vec<Felt252> = vec![Felt252::from_(invoke_response.transaction_hash.0)];
+                Ok(CheatcodeHandlingResult::Handled(res))
             }
             "get_nonce" => {
                 let block_id = as_cairo_short_string(&inputs[0])
                     .expect("Failed to convert entry point name to short string");
-                let account = self.runtime.block_on(get_account(
+                let account = self.tokio_runtime.block_on(get_account(
                     &self.config.account,
                     &self.config.accounts_file,
                     self.provider,
                     self.config.keystore.clone(),
                 ))?;
 
-                let nonce = self.runtime.block_on(get_nonce(
+                let nonce = self.tokio_runtime.block_on(get_nonce(
                     self.provider,
                     &block_id,
                     account.address(),
                 ))?;
-                buffer
-                    .write(Felt252::from_(nonce))
-                    .expect("Failed to insert nonce");
 
-                Ok(())
+                let res: Vec<Felt252> = vec![Felt252::from_(nonce)];
+                Ok(CheatcodeHandlingResult::Handled(res))
             }
-            _ => Err(anyhow!("Unknown cheatcode selector: {selector}")),
+            _ => Err(EnhancedHintError::from(anyhow!(
+                "Cheatcode {selector} not implemented"
+            ))),
         }?;
 
-        let result_end = buffer.ptr;
-        insert_value_to_cellref!(vm, output_start, result_start)?;
-        insert_value_to_cellref!(vm, output_end, result_end)?;
+        Ok(res)
+    }
 
-        Ok(())
+    fn override_system_call(
```

Comment:
why do we need this?

Reply:
To turn off Starknet syscalls. 
When we decide which ones we want to support, we will forward the call to blockifier for them and block others

Reply:
sure, thanks for clarification. Do you mind adding a comment about this?

Reply:
We could do that, but I'm afraid this comment can get outdated at some point.

I feel like it's deductible from the code since the name of the function suggests that we override syscall behaviour and for each syscall, the error is thrown (which tells that Starknet syscalls are not supported)

Reply:
I think the `override_system_call` may be renamed to `override_starknet_system_call` or `override_contract_system_call`

@MaksymilianDemitraszek WDYT?

Reply:
I think in case of plain cairo you just say "hints", I have seen word "syscall" used only in the context of starknet

Reply:
override_starknet_system_call i am ok with changing this if you think it will be more intuitive but please do it in a separate PR.